src/context/ThemeContext.tsx

```diff
@@ -0,0 +1,32 @@
+import { createContext, useEffect, useState } from "react";
+
+type ThemeContextType = {
+  theme: string;
+  toggleTheme: () => void;
+};
+export const ThemeContext = createContext<ThemeContextType | undefined>(
+  undefined
+);
+
+export default function ThemeContextProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [theme, setTheme] = useState(
+    () => localStorage.getItem("theme") || "light"
+  );
+  const toggleTheme = () => {
+    setTheme((prevTheme) => (prevTheme === "light" ? "dark" : "light"));
+  };
+  useEffect(() => {
+    document.documentElement.setAttribute("theme", theme);
+    localStorage.setItem("theme", theme);
```

Comment:
when having multiple taps open at the same time and changing the theme in one tap. An expected behviour is to have the theme changed in all taps. To achieve this, we can listen to the storage event and whenever the theme value changes, we update the UI.